tests/test_all.py

```diff
@@ -1,10 +1,16 @@
 import subprocess
 from unittest.mock import patch
-from pyp.pyp import pip_freeze
+from pyp.pyp import pip_freeze, sort_entries
 
 
 @patch('subprocess.check_output')
 def test_pip_freeze_search_patterns(mock_check_output):
     pip_freeze(['plain', 'with-hyphen', 'with_underscore', 'this-has_both'])
     mock_check_output.assert_called_once()
     mock_check_output.call_args[0][0] == "pip freeze | grep -i 'plain==\\|with[-_]hyphen==\\|with[-_]underscore==\\|this[-_]has[-_]both=='"
+
+def test_package_name_sorting_handles_irregular_names():
+    packages = ['Django-7.5', 'abc-33', 'rANndom_44', 'pip']
+    expected_output = ['abc-33', 'django-7.5', 'pip', 'ranndom_44']
+    sorted_packages = sort_entries(packages)
+    ''.join(sorted_packages) == ''.join(expected_output)
```

Comment:
Would you mind adding a newline to the end of this file

Reply:
Okay. I'm wondering why we're doing this, though? Python convention or conforming to the Unix file structure? Or maybe something else?

Reply:
I think having a newline at the end is a pep8 standard, and I've heard it can cause problems with running the script in some environments. Most likely it's fine either way, but I like to err on the safe side.